plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/CustomizeStore.php

```diff
@@ -18,7 +18,8 @@ public function __construct( $task_list ) {
 		parent::__construct( $task_list );
 
 		add_action( 'admin_enqueue_scripts', array( $this, 'possibly_add_site_editor_scripts' ) );
-		add_action( 'after_switch_theme', array( $this, 'mark_task_as_complete' ) );
+		// Use "switch_theme" instead of "after_switch_theme" because the latter is fired after the next WP load and we need to remove action when switching theme to TT3 via onboarding theme API. See Automattic\WooCommerce\Admin\API\OnboardingThemes::activate_theme().
+		add_action( 'switch_theme', array( $this, 'mark_task_as_complete' ) );
```

Comment:
Nitpick: is it possible to bring the "prevent" logic here?

```suggestion
		$theme_switch_via_cys_ai_loader = 1 === absint( $request['theme_switch_via_cys_ai_loader'] );
		if ( ! $theme_switch_via_cys_ai_loader ) {
				add_action( 'switch_theme', array( $this, 'mark_task_as_complete' ) );
		}
```

Reply:
Yes! 👍  Updated in 5a1c3c697c11c22608b3ef6d966e835ad2796fba.